### PR TITLE
refactor li proposal test 

### PR DESCRIPTION
**Pull request by jbibla**
- new test stylez for li proposal test

---

For contributor:

- [ ] Added entries in `CHANGELOG.md` with issue # and GitHub username
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: test/unit/specs/components/governance/LiProposal.spec.js [hunk 1]
```diff
@@ -86,29 +140,16 @@ describe(`LiProposal`, () => {
...
 
   it(`should truncate the description and add an ellipsis`, () => {
-    proposal.description = `this is some kind of long description. longer than 100 characters for optimum-maximum-ideal truncation.`
-    wrapper.setProps({ proposal: JSON.parse(JSON.stringify(proposal)) })
-    expect(wrapper.vm.description).toEqual(
-      `this is some kind of long description. longer than 100 characters for optimum-maximum-ideal truncati…`
-    )
-  })
-
-  it(`survives the tally result not being present yet`, () => {
```
Comment (faboweb) [inline test/unit/specs/components/governance/LiProposal.spec.js]: Why did you kill this test?

Comment (jbibla) [reply test/unit/specs/components/governance/LiProposal.spec.js]: it wasn't testing anything

Comment (faboweb) [reply test/unit/specs/components/governance/LiProposal.spec.js]: it probably tested, that it wouldn't throw an error.

File: test/unit/specs/components/governance/LiProposal.spec.js [hunk 2]
```diff
@@ -1,80 +1,134 @@
...
 
-const { proposals, tallies } = lcdClientMock.state
-const proposal = proposals[`2`]
-
-const $store = {
-  commit: jest.fn(),
-  dispatch: jest.fn(),
-  getters: {
-    proposals: { proposals, tallies }
+const proposals = {
```
Comment (sabau) [inline test/unit/specs/components/governance/LiProposal.spec.js]: I would move this away and import the same from every spec file that needs it, so if the payload changes, we fix a vuex module with his test file, we have to align all the ones that uses this payload with the new version of it

File: test/unit/specs/components/governance/LiProposal.spec.js [hunk 3]
```diff
@@ -1,80 +1,134 @@
...
   })
 
   it(`has the expected html structure`, () => {
     expect(wrapper.vm.$el).toMatchSnapshot()
   })
 
   it(`should return status info for passed proposals`, () => {
-    proposal.proposal_status = `Passed`
-    wrapper.setProps({ proposal: JSON.parse(JSON.stringify(proposal)) })
+    wrapper.setProps({
```
Comment (sabau) [inline test/unit/specs/components/governance/LiProposal.spec.js]: We could do something like: 
```
proposal = {stuff}
...
{
  proposal: {
    ...proposal,
   proposal_status: `the status I want to test`
  }
}
```

Comment (faboweb) [reply test/unit/specs/components/governance/LiProposal.spec.js]: same comment about cloning

File: CHANGELOG.md [hunk 4]
```diff
@@ -16,6 +16,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Changed
 
 - Refactored tests: PageStaking, PanelSort, TabStakingParameters @faboweb
+- Refactored tests: LiProposal @jbibla
```
Comment (fedekunze) [inline CHANGELOG.md]: this should go on `Unreleased`

File: test/unit/specs/components/governance/LiProposal.spec.js (hunk 3 above)
Comment (jbibla) [reply test/unit/specs/components/governance/LiProposal.spec.js]: i don't understand what you guys are suggesting.

File: test/unit/specs/components/governance/LiProposal.spec.js (hunk 2 above)
Comment (jbibla) [reply test/unit/specs/components/governance/LiProposal.spec.js]: ok will do.

File: test/unit/specs/components/governance/LiProposal.spec.js (hunk 1 above)
Comment (jbibla) [reply test/unit/specs/components/governance/LiProposal.spec.js]: ok i added a test back.

File: test/unit/specs/components/governance/LiProposal.spec.js (hunk 3 above)
Comment (faboweb) [reply test/unit/specs/components/governance/LiProposal.spec.js]: Karoly is proposing that you change the proposal instantiation. His way the difference between the proposals can be spotted quicker. It also limits the code duplication. Ignore my comment.

Comment (jbibla) [reply test/unit/specs/components/governance/LiProposal.spec.js]: done.

File: CHANGELOG.md [hunk 5]
```diff
@@ -47,6 +51,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### Fixed
 
+- Refactored tests: PageStaking, PanelSort, TabStakingParameters @faboweb
+
+## Fixed
```
Comment (fedekunze) [inline CHANGELOG.md]: dup fixed entry

File: app/src/renderer/components/governance/LiProposal.vue [hunk 6]
```diff
@@ -31,16 +31,16 @@
     </td>
     <td>{{ `#` + proposal.proposal_id }}</td>
     <td class="li-proposal__value yes">
-      {{ tally.yes }}
+      {{ tally.yes || `--` }}
```
Comment (fedekunze) [inline app/src/renderer/components/governance/LiProposal.vue]: side note: we should maybe prettify the tally result numbers

Comment (jbibla) [reply app/src/renderer/components/governance/LiProposal.vue]: what do you mean by prettify?

Comment (faboweb) [reply app/src/renderer/components/governance/LiProposal.vue]: there is a function called `pretty` which shortens the value after 4 decimals

**Pull request merged**